src/sysext/sysext.c

```diff
@@ -525,35 +698,79 @@ static int merge_hierarchy(
                 const char *meta_path,
                 const char *overlay_path) {
 
-        _cleanup_free_ char *resolved_hierarchy = NULL, *f = NULL, *buf = NULL;
+        _cleanup_free_ char *resolved_hierarchy = NULL, *f = NULL, *buf = NULL, *resolved_upper_dir = NULL;
         _cleanup_strv_free_ char **layers = NULL;
         struct stat st;
         int r;
+        bool drop_hierarchy = false;
 
         assert(hierarchy);
         assert(meta_path);
         assert(overlay_path);
 
+        f = extension_data_path_for_hierarchy(hierarchy);
+        if (f == NULL)
```

Comment:
as per coding style we generally prefer `if (!f)` to check for pointer validity.

Reply:
Fixed.